test/e2e/specs/site-editor/site-editor-url-navigation.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site editor url navigation', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'twentytwentyone' ),
+			// Fallback clean-up in case of failure.
+			requestUtils.deleteAllPosts(),
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.describe(
+		'Redirection after template and template part creation',
+		() => {
+			test( 'Redirection after template creation', async ( {
+				admin,
+				page,
+				requestUtils,
+			} ) => {
+				await requestUtils.createPost( {
+					title: 'Demo',
+					status: 'publish',
+				} );
+				await admin.visitSiteEditor();
+				// We need to wait the response from the `posts` request in order to click the augmented menu item,
+				// that includes the options for creating templates for specific posts.
+				await Promise.all( [
+					page.waitForResponse(
```

Comment:
Just being curious, do we have a UI hint for such things? Is waiting for the post list enough in this case?

Reply:
No we don't. This should be improved, but it's out of scope of this PR.